### PR TITLE
boards/openmoteb-cc2538: Initial support

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

Add the basic board files based off the original openmote board.

### Issues/PRs references

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
@PeterKietzmann I can't get the flasher working yet, maybe give it a try on your laptop...

**Comment by MrKevinWeiss** [comment]
I tested the UART, I2C, SPI and LEDs for the GPIO.  I think it is OK for the first basic port, other drivers will have to be added later.  Waited to be reviewed... I guess I should keep an eye on the commit message lengths (I never thought that I would be doing too much commenting).  Thanks to @A-Paul for figuring out the --bootloader-invert-lines to enable the bootloader.

**Comment by MrKevinWeiss** [comment]
Still sorting out some problems with the 2.4 GHz networking.  Maybe I will but @PeterKietzmann or @smlng

File: boards/openmoteb/board.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Copyright (C) 2014 Freie Universität Berlin
+ * Copyright (C) 2018 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     boards_openmoteb
```
Comment (smlng) [inline boards/openmoteb/board.c]: can we name it `boards_openmote_b` and the board directory should be `openmote-b` to be consistent with the rest

**Comment by jia200x** [comment]
Tested and works OK. Please squash

**Comment by smlng** [comment]
IMHO it makes sense to introduce a `common/openmote` directory to avoid code and config duplication - see for instance `common/remote` as a reference.

**Comment by MrKevinWeiss** [comment]
@smlng Sounds like a good idea but I think it should be a different PR.  Maybe merge this one then I will do that next.

**Comment by smlng** [comment]
@MrKevinWeiss I'm fine with doing that in a separate PR, is this ready for merging? Do we have the hardware for testing (yet)?

**Comment by MrKevinWeiss** [comment]
Yup as long a murdock is happy!  It is tested on hardware (and i2c has been retested).

**Comment by MrKevinWeiss** [comment]
@aabadie now that the embargo is over can we merge?

**Comment by PeterKietzmann** [comment]
@jia200x, @aabadie please remove your change requests - if you are fine with this PR.

File: boards/openmote-b/Makefile.features [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_gpio
+FEATURES_PROVIDED += periph_i2c
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_timer
+FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_adc
```
Comment (aabadie) [inline boards/openmote-b/Makefile.features]: This line should be moved to the top to keep alphabetical order.

File: boards/openmote-b/include/periph_conf.h [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        .rts_pin  = GPIO_UNDEF
+    }
+};
+
+/* interrupt function name mapping */
+#define UART_0_ISR          isr_uart0
+
+/* macros common across all UARTs */
+#define UART_NUMOF          (sizeof(uart_config) / sizeof(uart_config[0]))
+
```
Comment (aabadie) [inline boards/openmote-b/include/periph_conf.h]: This blank line can be removed

**Comment by MrKevinWeiss** [comment]
@aabadie oh how beautiful it looks with alphabetical order and no extra white lines, would you like to do the honors?

**Comment by smlng** [comment]
🎉

**Comment by PeterKietzmann** [comment]
Congrats!

**Comment by jasonatran** [comment]
@MrKevinWeiss, thanks for the initial support contribution. We over at USC have been using the old OpenMotes  for research and teaching for quite some time. We recently had to order the new openmote-b's to scale up a course, but the communication range really short (as if the antenna isn't working). Any ideas?

**Comment by MrKevinWeiss** [comment]
@jasonatran I did notice this during the initial port but then tested on all cc2538 platforms and found the same issues (so it was a problem with the driver and not the board).  There is a switch that allows the internal and external antenna to be selected, you could try changing that value. I will be able to confirm on the current master tomorrow (27.09.2018 @ 09:00 CET).

**Comment by PeterKietzmann** [comment]
@jasonatran is this behavior specific to RIOT or do you face similar problems with OpenWSN for example? This PR is probably not the right place for this topic. If it relates to RIOT, would you open an issue? Furthermore, @changtengfei  might give some insights. To be a more precise: The issue relates to the 2,4 GHz radio

**Comment by changtengfei** [comment]
@jasonatran @PeterKietzmann , on the openmote-b there are two radios, cc2538 (2.4ghz)and at86rf215 (2.4ghz and subghz). There is only one 2.4ghz antenna, it uses a logic unit to select which radio to use the antenna and controlled by cc2538.  The right logic to configure cc2538 to use the 2.4ghz antenna is PD4 low and PD3 high. You check check whether this is the current setting.

**Comment by PeterKietzmann** [comment]
Merci @changtengfei, good point! If I see it correctly this is currently undefined. I'll investigate

**Comment by MrKevinWeiss** [comment]
Are you talking about what happens in the board_init?

```
    gpio_init(RF_SWITCH_2_4_GHZ_PIN, GPIO_OUT);
    gpio_init(RF_SWITCH_SUB_GHZ_PIN, GPIO_OUT);
     /* start with cc2538 2.4ghz radio*/
    RF_SWITCH_2_4_GHZ_ON;
    RF_SWITCH_SUB_GHZ_OFF;
```

**Comment by PeterKietzmann** [comment]
Oh, yes I am. Accidentally checked the init function of openmote-2535 :/

**Comment by PeterKietzmann** [comment]
Wait a second. Tengfei said PD4 **low** and PD3 **high** for 2,4GHz antenna. If I see it correctly we do it the other way around, see [here](https://github.com/RIOT-OS/RIOT/blob/master/boards/openmote-b/include/board.h#L72-L78). @jasonatran mind to give it a try?

**Comment by jasonatran** [comment]
@PeterKietzmann this is specific to RIOT. @changtengfei (hey!) thanks for the info. Changing PD4 to low and PD3 high worked. i.e. I used these lines in board.c:

`RF_SWITCH_2_4_GHZ_OFF;`
`RF_SWITCH_SUB_GHZ_ON;`

**Comment by changtengfei** [comment]
Hi @jasonatran , great to hear you  again! I'm glade it works!

**Pull request merged**